utils/check-for-deprecations.js

```diff
@@ -0,0 +1,199 @@
+const rawGlobalsdata = fs.readFileSync("../globals.json");
+const globals = JSON.parse(rawGlobalsdata);
+
+const pullRequestIntro =
+  "## Description \nThe following apps and deprecation messages have been logged in the platform. To prevent users of these apps seeing error toasts in the platform, please make the necessary changes.\nSee for more context:\n - https://docs.google.com/document/d/1kyaxHKxVqTcyaayKK3TG6MLXvje3Uc59RqX1kC9bjE0/edit#\n - https://nerdlife.datanerd.us/new-relic/logger-improvements-834d24dc-16c0-4e64-b6a1-95c03100e779\n\n## Apps\n";
+const timeframe = "1 month";
+const retries = 5;
+const ACCOUNT_ID = process.env.NR_ACCOUNT_ID;
+const queryKey = process.env.NR_QUERY_KEY;
+const basePath = `https://staging-insights-api.newrelic.com/v1/accounts/${ACCOUNT_ID}/query?nrql=`;
+const options = {
+  headers: {
+    "X-Query-Key": queryKey,
+  },
+};
+
+const context = github.context;
+const octokit = new Octokit({
+  auth: process.env.GITHUB_TOKEN,
+});
+
+/**
+ * Stringifies an array for use in a NRQL query
+ * @param {Array} arr - An array of appName strings
+ * @returns {String} stringifiedArray
+ */
+const stringifyArrayForNrql = (arr) => {
+  const stringifiedArray = Object.values(arr).reduce((acc, name) => {
+    acc += ",";
+    return (acc += `'${name}'`);
+  }, "");
+
+  return stringifiedArray;
+};
+
+/**
+ * Fetches event data from the Insights Event API
+ * @param {String} url - The url to send the request to
+ * @param {Object} options - The request options
+ * @returns {Object} response.json()
+ */
+const getData = async (url, options) => {
+  try {
+    const response = await fetch(url, options);
+
+    if (response.status >= 300) {
+      throw new pRetry.AbortError({
+        errorMessage: `Request failed after ${retries} retries`,
+        statusText: response.statusText,
+        status: response.status,
+        retries,
+        url,
+      });
+    }
+    return response.json();
+  } catch (err) {
+    core.setFailed("Action failed with error:");
+    core.error(JSON.stringify(err));
+    return [];
+  }
+};
+
+/**
+ * Replaces %s in messages with an array of arguments
+ * @param {Object} events - Events from API containing message and args
+ * @returns {Array} An array of unique messages with args replaced
+ */
+const replaceMessageArgs = async (events) => {
+  const replacedMessages = await Promise.all(
+    events.map(async (event) => {
+      const args = JSON.parse(event.args);
+
+      let message = await args.reduce((acc, arg) => {
+        acc = acc.replace(/%s/, arg);
+        return acc;
+      }, event.message);
+
+      return message;
+    })
+  );
+
+  const uniqueMessages = await getUniqueMessages(replacedMessages);
+  return uniqueMessages;
+};
+
+/**
+ * Removes duplicates from an array
+ * @param {Array} messages - Messages with args replaced
+ * @returns {Array} An array of Unique Messages
+ */
+const getUniqueMessages = async (messages) => {
+  const uniqueMessages = await Promise.all(
```

Comment:
I don't think you need this `await Promise.all` since the underlying function doesn't return a promise. This appears to be fully synchronous

Reply:
Ye sure lets chat about these whenever you're online :D